asdf/_tests/_issues/test_1013.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+import asdf
+
+
+def test_1013(tmp_path):
```

Comment:
As noted in https://github.com/asdf-format/asdf/pull/1537#issuecomment-1618994330. I strongly disagree with naming tests after issues. I also disagree with your organization method of each issue being its own module. The reason being that the test modules are "supposed" to be some reasonable categorization of like tests. For example, the `test_manager` module should contain most of the tests which will "smoke" out any issues related to the "manager". This way when working on a specific section of the code base the developer can iterate much faster against the tests categorized to be related to what they are doing. This becomes ever more important the longer the test suite takes to run.

This naming scheme both destroys this categorization AND as I already commented, makes what tests are supposed to be testing extremely opaque. At the very least it should not be both, and I think the test name is more important than the module name. However, if you want to go this way I believe you should introduce categorical test markers to so that the tests can be reduced to specific categories.

Since the "issue" categorization is useful I think things should be organized via test classes instead. By this I mean:
```python
class Test1013:
    """documentation linking to #1013 and a basic description"""

    # Specifically named tests concerning the solution to #1013
    ...
```
These test classes can then be incorporated into the categorical modules that already exist or in new ones as needed. By doing this you largely cover all the major concerns:

1. Issues being directly linked to tests.
2. Descriptive test names.
3. Test categorization.

I do grant that the current test categorization/modules do leave much to be desired, and so I fully support reorganizing those and their tests into better "categories".